src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java

```diff
@@ -99,8 +101,17 @@ public void collectOpenshiftMetrics(String account, OffsetDateTime start, Offset
                     );
                 }
 
+                Map<EventKey, Event> existing = eventController.mapEventsInTimeRange(
+                    account,
+                    MeteringEventFactory.OPENSHIFT_CLUSTER_EVENT_SOURCE,
+                    MeteringEventFactory.OPENSHIFT_CLUSTER_EVENT_TYPE,
+                    start,
+                    end
+                );
+                log.debug("Found {} existing events.", existing.size());
+
                 // Given the possibility of a very large number of events, we will batch persist them
-                // to provide the ability to tweak them.
+                // to provide the ability to tweak the size later.
                 List<Event> events = new LinkedList<>();
```

Comment:
Rather than use a `List` here, for consistency, I think you should use `Map<EventKey, Event>`.  That way, we can handle the case where there are conflicting events in the same batch from Prometheus (last-in wins). I think we'll also have to drop the batching mechanism for this (though I think it likely wasn't doing too much for us since this is all in one transaction anyways).

Reply:
DONE